src/controllers/product.js

```diff
@@ -40,6 +41,38 @@ const addNewProduct = async (req, res) => {
   }
 };
 
+const deleteProduct = async (req, res) => {
+  try {
+    const singlePost = await ProductModel.findById(req.params.id);
+
+    if (singlePost) {
```

Comment:
if we agreed not to delete the received product I think we should check for the isTransactionCompleted field too.